### PR TITLE
Model edit refactor

### PR DESCRIPTION
**Pull request by jchang691**
When a model is edited multiple times, it will no longer go through already edited values.

### CODE REVIEW

File: lib/watirmark/models/factory.rb [hunk 1]
```diff
@@ -14,7 +14,7 @@ class Factory
       include DebugMethods
       include FactoryMethodGenerators
 
-      attr_accessor :defaults, :model_name, :models, :parent, :children, :model_type
+      attr_accessor :defaults, :model_name, :models, :parent, :children, :model_type, :updates
```
Comment (jchang691) [inline lib/watirmark/models/factory.rb]: Remove from accessor.

File: spec/model_factory_spec.rb [hunk 2]
```diff
@@ -99,6 +99,20 @@ class UpdateModel < Watirmark::Model::Factory
     lambda{keys.update('   '.to_sym=>'') }.should_not raise_error
   end
 
+  specify "model #updates hash is updated when any keywords in the model is updated" do
+    keys = FactoryTest::UpdateModel.new
+    keys.update(:username=>'username', :foo=>'foo')
+    expect(keys.updates.include? :username).to be true
+    expect(keys.updates.include? :foo).to be true
```
Comment (jchang691) [inline spec/model_factory_spec.rb]: change to expect(keys.updates).to include? Foo

**Comment by jchang691** [comment]
Add a log line in the edit to show what is going to populated.

**Comment by jchang691** [comment]
get_data_model edit. Check this out if this edit/update is backward compatible.

**Comment by Blackbaud-KyleMartinez** [comment]
Add test to verify backward compatibility

**Pull request merged**